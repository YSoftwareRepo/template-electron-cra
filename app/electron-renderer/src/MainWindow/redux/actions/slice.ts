import { createSlice } from '@reduxjs/toolkit';
import { omitUnderscorePrefixActions } from '@tecra-pkg/utils-redux';

import { StoreState } from '../types';
import { _finishPreparing, _setAppDetails } from './reducers';
import { updateClientAreaSize } from './reducers.DisplayConfig';

export function getInitialState(): StoreState {
    return {
        appDetails: null as never,
        displayConfig: {
            clientAreaSize: null as never,
        },
        prepared: false,
    };
}

const slice = createSlice({
    name: 'Main',
    initialState: getInitialState(),
    reducers: {
        _finishPreparing,
        _setAppDetails,
        updateClientAreaSize,
    },
});

export const _actions = slice.actions;
export const _reducer = slice.reducer;

export const actions = omitUnderscorePrefixActions(slice.actions);
