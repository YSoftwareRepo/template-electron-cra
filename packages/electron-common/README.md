# @tecra-pkg/electron-common

Common code for electron main process and electron renderer process.
