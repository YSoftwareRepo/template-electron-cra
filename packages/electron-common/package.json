{
    "name": "@tecra/electron-common",
    "version": "0.8.0",
    "description": "Common code for electron main process and electron renderer process.",
    "main": "lib/index.js",
    "module": "lib/index.js",
    "typings": "lib/index.d.ts",
    "homepage": "https://github.com/YSoftwareRepo/template-electron-cra#readme",
    "license": "MIT",
    "author": "yifanwww <yifanw1101@gmail.com> (https://github.com/yifanwww)",
    "bugs": {
        "url": "https://github.com/YSoftwareRepo/template-electron-cra/issues"
    },
    "private": true,
    "repository": {
        "type": "git",
        "url": "git+https://github.com/YSoftwareRepo/template-electron-cra.git",
        "directory": "packages/electron-common"
    },
    "scripts": {
        "build": "concurrently -n esm,types \"pnpm run build:esm\" \"pnpm run build:types\"",
        "build:esm": "tsc --project tsconfigs/tsconfig.esm.json",
        "build:types": "tsc --project tsconfigs/tsconfig.types.json",
        "clean": "rimraf coverage lib tsconfigs/*.tsbuildinfo",
        "dev": "pnpm run build:esm -- --watch",
        "format": "format-code",
        "lint": "eslint .",
        "pre-commit": "lint-staged"
    },
    "devDependencies": {
        "@tecra/eslint-config": "workspace:*",
        "@tecra/global-types": "workspace:*",
        "@tecra/scripts": "workspace:*",
        "@tecra/tsconfigs": "workspace:*"
    }
}
