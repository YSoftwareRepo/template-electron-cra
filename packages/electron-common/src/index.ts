export * from './apis';
export * from './timer';
export * from './type';
export * from './utils';
