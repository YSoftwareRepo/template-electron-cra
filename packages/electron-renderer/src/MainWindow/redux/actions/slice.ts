import { createSlice } from '@reduxjs/toolkit';
import { omitUnderscorePrefixActions } from '@tecra/utils-redux';

import { IStoreState } from '../types';
import { _finishPreparing, _setAppDetails } from './reducers';
import { updateClientAreaSize } from './reducers.DisplayConfig';

export function getInitialState(): IStoreState {
    return {
        appDetails: null as never,
        displayConfig: {
            clientAreaSize: null as never,
        },
        prepared: false,
    };
}

const slice = createSlice({
    name: 'Main',
    initialState: getInitialState(),
    reducers: {
        _finishPreparing,
        _setAppDetails,
        updateClientAreaSize,
    },
});

export const _actions = slice.actions;
export const _reducer = slice.reducer;

export const actions = omitUnderscorePrefixActions(slice.actions);
