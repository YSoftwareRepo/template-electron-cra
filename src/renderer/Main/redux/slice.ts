import { createSlice } from '@reduxjs/toolkit';

import type { ClientAreaSize, IReducer, IMapActionsToProps } from '#RUtils/Types';

interface MainState {
    clientAreaSize: ClientAreaSize;
}

const initialState: MainState = {
    clientAreaSize: { width: 1280, height: 720 },
};

// ---------------------------------------------------------------------------------------- Reducers

type Reducer<Payload = undefined> = IReducer<MainState, Payload>;

const updateClientAreaSize: Reducer<ClientAreaSize> = (state, action) => {
    state.clientAreaSize = action.payload;

    console.log(`update clientarea: ${JSON.stringify(action.payload)}.`);
};

// ------------------------------------------------------------------------------------------- Slice

export const { actions: mainActions, reducer: mainReducer } = createSlice({
    name: 'ReactApp',
    initialState,
    reducers: {
        updateClientAreaSize,
    },
});

export type MainMapActionsToProps = IMapActionsToProps<typeof mainActions>;
