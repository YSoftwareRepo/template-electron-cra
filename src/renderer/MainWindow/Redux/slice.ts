import { configureStore, createSlice } from '@reduxjs/toolkit';

import { IClientAreaSize } from '#RUtils/GlobalTypes';
import { IReducer } from '#RUtils/Redux';

import { StoreState } from './types';

const initialState: StoreState = {
    clientAreaSize: { width: 1280, height: 720 },
};

// ------------------------------------------------------------------------------------------------------------ Reducers

type Reducer<Payload = undefined> = IReducer<StoreState, Payload>;

const updateClientAreaSize: Reducer<IClientAreaSize> = (state, action) => {
    state.clientAreaSize = action.payload;
};

// --------------------------------------------------------------------------------------------------------------- Slice

const { actions, reducer } = createSlice({
    name: 'Main',
    initialState,
    reducers: {
        updateClientAreaSize,
    },
});

export { actions };

export const store = configureStore({ reducer });
