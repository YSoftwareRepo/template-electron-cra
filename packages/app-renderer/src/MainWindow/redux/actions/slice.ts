import { createSlice } from '@reduxjs/toolkit';

import { omitUnderscorePrefixActions } from 'src/utils/redux';

import type { StoreState } from '../types';

import { _setAppDetails } from './reducers.app';

export function getInitialState(): StoreState {
    return {};
}

const slice = createSlice({
    name: 'Main',
    initialState: getInitialState(),
    reducers: {
        _setAppDetails,
    },
});

export const _actions = slice.actions;
export const _reducer = slice.reducer;

export const actions = omitUnderscorePrefixActions(slice.actions);
