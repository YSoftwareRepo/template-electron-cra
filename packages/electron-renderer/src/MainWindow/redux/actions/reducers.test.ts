import { IAppDetails } from '@tecra/electron-common';
import { getInitialState, _actions, _reducer } from './slice';

describe('Test redux reducer `_setAppDetails`', () => {
    test('finishes preparing', () => {
        const prevState = getInitialState();

        const appDetails: IAppDetails = {
            name: 'tecra',
            version: 'unknown',
            module: {
                chrome: 'unknown',
                electron: 'unknown',
                node: 'unknown',
                v8: 'unknown',
            },
        };

        expect(_reducer(prevState, _actions._setAppDetails(appDetails)).appDetails).toMatchSnapshot();
    });
});

describe('Test redux reducer `_finishPreparing`', () => {
    test('finishes preparing', () => {
        const prevState = getInitialState();

        expect(_reducer(prevState, _actions._finishPreparing()).prepared).toMatchSnapshot();
    });
});
