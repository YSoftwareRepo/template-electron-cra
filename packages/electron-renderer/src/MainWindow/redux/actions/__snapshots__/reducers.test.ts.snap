// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test redux reducer \`_finishPreparing\` finishes preparing 1`] = `true`;

exports[`Test redux reducer \`_setAppDetails\` finishes preparing 1`] = `
Object {
  "module": Object {
    "chrome": "unknown",
    "electron": "unknown",
    "node": "unknown",
    "v8": "unknown",
  },
  "name": "tecra",
  "version": "unknown",
}
`;
